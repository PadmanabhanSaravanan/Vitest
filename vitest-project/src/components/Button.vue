<!-- src/components/Button.vue -->
<template>
  <button class="btn primary">{{ label }}!</button>  <!-- Added '!' and new class -->
</template>

<script setup>
defineProps({ label: String });
</script>