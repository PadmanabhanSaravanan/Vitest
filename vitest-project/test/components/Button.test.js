/// tests/components/Button.test.js
import { mount } from '@vue/test-utils';
import { describe, it, expect } from 'vitest';
import Button from '../../src/components/Button.vue';

describe('Button Component', () => {
  it('matches the snapshot', () => {
    const wrapper = mount(Button, {
      props: { label: 'Click Me' },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});