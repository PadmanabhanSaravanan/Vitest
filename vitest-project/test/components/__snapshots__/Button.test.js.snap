// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button Component > matches the snapshot 1`] = `
"<button class="btn primary">Click Me!</button>
<!-- Added '!' and new class -->"
`;
